server/src/capture.rs

```diff
@@ -0,0 +1,198 @@
+use log::{error, info};
+use serde::Deserialize;
+use simplelog::*;
+use simplelog::{ColorChoice, TermLogger, TerminalMode};
+use std::fs;
+use std::fs::File;
+use std::path::Path;
+use tokio::io::{AsyncBufReadExt, BufReader};
+use tokio::net::TcpListener;
+use tokio::sync::mpsc;
+use tokio_postgres::{Client, NoTls}; // Import necessary types
+
+#[derive(Debug, Deserialize)]
+struct DatabaseConfig {
+    db_name: String,
+    db_ip: String,
+    db_user: String,
+    db_password: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct Event {
+    user_id: String,
+    event_type: String,
+    timestamp: String,
+    data: Option<String>,
+}
+
+fn load_config() -> Result<DatabaseConfig, Box<dyn std::error::Error>> {
+    // Specify the path to your configuration file
+    let config_path = Path::new("config.json");
```

Comment:
Make this independent of the working directly, so the user can start from anywhere. See `ROOT_PATH` in `webserver.rs`.

Reply:
The function has been removed and the code rolled into the EventCapture new